dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/RefinementSolver.java

```diff
@@ -560,15 +559,15 @@ private static void addBiases(Wmm wmm, EnumSet<Baseline> biases) {
         final Relation rfinv = wmm.addDefinition(new Inverse(wmm.newRelation(), rf));
         final Relation frStandard = wmm.addDefinition(new Composition(wmm.newRelation(), rfinv, co));
 
-        // ([R] \ [range(rf)]);loc;[W]
-        final Relation reads = wmm.addDefinition(new SetIdentity(wmm.newRelation(), wmm.getFilter(Tag.READ)));
-        final Relation rfRange = wmm.addDefinition(new RangeIdentity(wmm.newRelation(), rf));
-        final Relation writes = wmm.addDefinition(new SetIdentity(wmm.newRelation(), Filter.byTag(Tag.WRITE)));
-        final Relation ur = wmm.addDefinition(new Difference(wmm.newRelation(), reads, rfRange));
-        final Relation urloc = wmm.addDefinition(new Composition(wmm.newRelation(), ur, loc));
-        final Relation urlocwrites = wmm.addDefinition(new Composition(wmm.newRelation(), urloc, writes));
+        // ((R \ range(rf)) * W) & loc
```

Comment:
I'm not convinced that this formulation is better, because the lhs of the intersection is now always large.
I think the original one was fine, and all you need to do is move the difference inside the set identity "[R] \ [range(rf)] --> [R \ range(rf)]".

Reply:
I agree.  This change's purpose was for debugging.  It should be as you described now.